### PR TITLE
DOC: Fix format of basics.rst, following PEP-8 standard

### LINKED ISSUE #23792: DOC: Fix format of basics.rst
Our documentation should follow the PEP-8 standard, but for historical reasons there are many formats that don't. We should fix them now.

This issue is to fix the format of `doc/source/basics.rst`. To find the errors, first is required to install a newer version of `flake8-rst`:
```
conda install -c conda-forge flake8-rst=0.5.0
```

Then, all the errors can be obtained with:
```
$ flake8-rst doc/source/basics.rst 
doc/source/basics.rst:309:10: E225 missing whitespace around operator
doc/source/basics.rst:6:4: E902 TokenError: EOF in multi-line statement
doc/source/basics.rst:152:28: E203 whitespace before ':'
doc/source/basics.rst:152:83: E501 line too long (80 > 79 characters)
doc/source/basics.rst:153:28: E203 whitespace before ':'
doc/source/basics.rst:153:83: E501 line too long (85 > 79 characters)
doc/source/basics.rst:154:30: E203 whitespace before ':'
doc/source/basics.rst:154:83: E501 line too long (83 > 79 characters)
doc/source/basics.rst:175:46: E231 missing whitespace after ','
doc/source/basics.rst:175:51: E231 missing whitespace after ','
doc/source/basics.rst:175:54: E231 missing whitespace after ','
doc/source/basics.rst:175:59: E231 missing whitespace after ','
doc/source/basics.rst:175:62: E231 missing whitespace after ','
doc/source/basics.rst:175:67: E231 missing whitespace after ','
doc/source/basics.rst:175:70: E231 missing whitespace after ','
doc/source/basics.rst:176:57: E231 missing whitespace after ','
doc/source/basics.rst:339:6: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:339:15: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:340:7: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:340:16: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:355:7: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:355:21: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:362:29: E231 missing whitespace after ':'
doc/source/basics.rst:363:29: E231 missing whitespace after ':'
doc/source/basics.rst:363:59: E231 missing whitespace after ','
doc/source/basics.rst:363:61: E231 missing whitespace after ','
doc/source/basics.rst:426:27: E203 whitespace before ':'
doc/source/basics.rst:427:27: E203 whitespace before ':'
doc/source/basics.rst:428:27: E203 whitespace before ':'
doc/source/basics.rst:429:27: E203 whitespace before ':'
doc/source/basics.rst:446:4: E731 do not assign a lambda expression, use a def
doc/source/basics.rst:549:17: E221 multiple spaces before operator
doc/source/basics.rst:566:84: E501 line too long (81 > 79 characters)
doc/source/basics.rst:622:40: E231 missing whitespace after ','
doc/source/basics.rst:622:57: E231 missing whitespace after ','
doc/source/basics.rst:622:61: E231 missing whitespace after ','
doc/source/basics.rst:751:4: E402 module level import not at top of file
doc/source/basics.rst:756:19: E251 unexpected spaces around keyword / parameter equals
doc/source/basics.rst:756:21: E251 unexpected spaces around keyword / parameter equals
doc/source/basics.rst:760:4: E124 closing bracket does not match visual indentation
doc/source/basics.rst:883:83: E501 line too long (86 > 79 characters)
doc/source/basics.rst:927:7: E111 indentation is not a multiple of four
doc/source/basics.rst:980:4: E402 module level import not at top of file
doc/source/basics.rst:1041:39: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1048:41: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1059:49: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1067:50: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1087:4: E731 do not assign a lambda expression, use a def
doc/source/basics.rst:1099:24: E203 whitespace before ':'
doc/source/basics.rst:1099:38: E203 whitespace before ':'
doc/source/basics.rst:1115:4: E402 module level import not at top of file
doc/source/basics.rst:1124:36: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1152:15: E126 continuation line over-indented for hanging indent
doc/source/basics.rst:1152:27: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1152:37: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1162:4: E731 do not assign a lambda expression, use a def
doc/source/basics.rst:1162:23: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1162:34: E226 missing whitespace around arithmetic operator
doc/source/basics.rst:1164:32: E251 unexpected spaces around keyword / parameter equals
doc/source/basics.rst:1164:34: E251 unexpected spaces around keyword / parameter equals
doc/source/basics.rst:1164:43: E231 missing whitespace after ','
doc/source/basics.rst:1166:16: E231 missing whitespace after ','
doc/source/basics.rst:1166:18: E231 missing whitespace after ','
doc/source/basics.rst:1172:28: E201 whitespace after '['
doc/source/basics.rst:1172:47: E231 missing whitespace after ','
doc/source/basics.rst:1172:49: E231 missing whitespace after ','
doc/source/basics.rst:1173:28: E128 continuation line under-indented for visual indent
doc/source/basics.rst:1173:54: E202 whitespace before ']'
doc/source/basics.rst:1175:16: E231 missing whitespace after ','
doc/source/basics.rst:1175:18: E231 missing whitespace after ','
doc/source/basics.rst:1505:38: E203 whitespace before ':'
doc/source/basics.rst:1505:67: E203 whitespace before ':'
doc/source/basics.rst:1628:34: E231 missing whitespace after ','
doc/source/basics.rst:1679:14: E225 missing whitespace around operator
doc/source/basics.rst:1775:28: E203 whitespace before ':'
doc/source/basics.rst:1775:83: E501 line too long (80 > 79 characters)
doc/source/basics.rst:1776:28: E203 whitespace before ':'
doc/source/basics.rst:1776:83: E501 line too long (85 > 79 characters)
doc/source/basics.rst:1777:30: E203 whitespace before ':'
doc/source/basics.rst:1777:83: E501 line too long (83 > 79 characters)
doc/source/basics.rst:1803:29: E231 missing whitespace after ':'
doc/source/basics.rst:1803:32: E231 missing whitespace after ','
doc/source/basics.rst:1803:34: E231 missing whitespace after ','
doc/source/basics.rst:1803:36: E231 missing whitespace after ','
doc/source/basics.rst:1803:39: E231 missing whitespace after ','
doc/source/basics.rst:1803:45: E231 missing whitespace after ':'
doc/source/basics.rst:1803:48: E231 missing whitespace after ','
doc/source/basics.rst:1803:50: E231 missing whitespace after ','
doc/source/basics.rst:1803:52: E231 missing whitespace after ','
doc/source/basics.rst:1803:55: E231 missing whitespace after ','
doc/source/basics.rst:1803:63: E231 missing whitespace after ':'
doc/source/basics.rst:1803:66: E231 missing whitespace after ','
doc/source/basics.rst:1803:68: E231 missing whitespace after ','
doc/source/basics.rst:1803:70: E231 missing whitespace after ','
doc/source/basics.rst:1810:54: E231 missing whitespace after ','
doc/source/basics.rst:1913:49: E231 missing whitespace after ','
doc/source/basics.rst:1913:56: E231 missing whitespace after ','
doc/source/basics.rst:1913:61: E231 missing whitespace after ','
doc/source/basics.rst:1913:68: E231 missing whitespace after ','
doc/source/basics.rst:1913:73: E231 missing whitespace after ','
doc/source/basics.rst:1913:83: E501 line too long (80 > 79 characters)
doc/source/basics.rst:1914:27: E231 missing whitespace after ','
doc/source/basics.rst:1986:5: E999 SyntaxError: invalid syntax
doc/source/basics.rst:2010:83: E501 line too long (85 > 79 characters)
doc/source/basics.rst:2124:85: E501 line too long (83 > 79 characters)
doc/source/basics.rst:2212:84: E501 line too long (81 > 79 characters)
```

After fixing this issue, the previous command should not report any error.

### PR DESCRIPTION
**Pull request by ywpark1**
- [x] closes #23792 
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I updated the format of `doc/source/basics.rst` file. 
I do not see any errors when I run `flake8-rst doc/source/basics.rst`.

Please let me know if there are some changes.

### CODE REVIEW

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: I don't understand why we're adding a white space after all directives. Is there a reason, or is it your editor that did those automatically?

Afaik this space is wrong, and they should be removed again.

**Review by datapythonista — CHANGES_REQUESTED** [review]
Will review in detail after the spaces are removed, as they are wrong and they are making the review much more difficult.

**Comment by ywpark1** [comment]
I removed the spaces, and it shows these errors :
```
doc/source/basics.rst:310:10: E225 missing whitespace around operator
doc/source/basics.rst:6:4: E902 TokenError: EOF in multi-line statement
doc/source/basics.rst:754:4: E402 module level import not at top of file
doc/source/basics.rst:984:4: E402 module level import not at top of file
doc/source/basics.rst:1120:4: E402 module level import not at top of file
doc/source/basics.rst:1992:5: E999 SyntaxError: invalid syntax
```

Did I fix something wrong?

**Review by datapythonista — COMMENTED** [review]
Looks great. I commented on couple of extra things that I'd fix, but it's a nice improvement.

**Comment by datapythonista** [comment]
I can't see why you get these errors, but don't worry about them, we can merge all the fixes you made, and check later whether they are caused by a bug in flake-rst or what's wrong.

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: Do you mind changing this by `print(row_index, row, sep='\n')`

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: could you move `G` to thenext line for consistency and clairity?

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: If the `...` is giving problems with the validation I'd replace it by a `pass`, and we can also remove the `noqa` comment then.

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: just a personal opinion here, but I think this style is easier to read:
```
df = pd.DataFrame({'col1': np.random.randn(3),
                   'col2': np.random.randn(3)},
```

**Comment by ywpark1** [comment]
I fixed the missing spaces, adjusted indentation, and removed 'In [0]' from ipython.

I also fixed other errors occurred after changes. When I run `flake8-rst doc/source/basics.rst`, I see only these errors :
```
doc/source/basics.rst:753:4: E402 module level import not at top of file
doc/source/basics.rst:983:4: E402 module level import not at top of file
doc/source/basics.rst:1119:4: E402 module level import not at top of file
doc/source/basics.rst:2128:6: E402 module level import not at top of file
doc/source/basics.rst:2158:6: E402 module level import not at top of file
doc/source/basics.rst:2178:5: E402 module level import not at top of file
doc/source/basics.rst:2193:5: E402 module level import not at top of file
doc/source/basics.rst:2219:5: E402 module level import not at top of file
```

File: doc/source/basics.rst
Comment (datapythonista) [inline doc/source/basics.rst]: I think this is clearer here, and should fit in 79 characters:
```
C=pd.Series(np.array(np.random.randn(8),
                     dtype='uint8'))))
```

**Review by datapythonista — COMMENTED** [review]
Nice changes. Added few suggestions on things that I think can be presented more clearly, but looks good.

**Comment by datapythonista** [comment]
That's cool, thanks @ywpark1. I think the `E402 module level import not at top of file` shouldn't be reported by `flake8-rst`.

I'd add it to the list of flake8-rst errors to ignore in `setup.cfg`.

@FHaase what do you think?

**Review by datapythonista — APPROVED** [review]
Looks great, thanks @ywpark1

**Comment by ywpark1** [comment]
Hi @datapythonista ,
If you do not mind, can I try to re-open this Pull Request with the same changes? I touched something wrong in the local git and now it shows bunch of weird commits showing on this Pull Request.

File: doc/source/basics.rst
Comment (jorisvandenbossche) [inline doc/source/basics.rst]: I don't directly see how the above changes cause this, but, this example is now failing in the doc build with:

```
>>>-------------------------------------------------------------------------
Exception in /home/travis/build/pandas-dev/pandas/doc/source/basics.rst at block ending on line 1183
Specify :okexcept: as an option in the ipython:: block to suppress this message

KeyErrorTraceback (most recent call last)

~/build/pandas-dev/pandas/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   3146             try:
-> 3147                 return self._engine.get_loc(key)
   3148             except KeyError:
~/build/pandas-dev/pandas/pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()
~/build/pandas-dev/pandas/pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()
~/build/pandas-dev/pandas/pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()
~/build/pandas-dev/pandas/pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()
KeyError: 'ItemA'

During handling of the above exception, another exception occurred:

KeyErrorTraceback (most recent call last)
<ipython-input-209-ce1c0f9e4a44> in <module>
----> 1 result.loc[:, :, 'ItemA']
~/build/pandas-dev/pandas/pandas/core/indexing.py in __getitem__(self, key)
   1495             except (KeyError, IndexError, AttributeError):
   1496                 pass
-> 1497             return self._getitem_tuple(key)
   1498         else:
   1499             # we by definition only have the 0th axis
~/build/pandas-dev/pandas/pandas/core/indexing.py in _getitem_tuple(self, tup)
    868     def _getitem_tuple(self, tup):
    869         try:
--> 870             return self._getitem_lowerdim(tup)
    871         except IndexingError:
    872             pass
~/build/pandas-dev/pandas/pandas/core/indexing.py in _getitem_lowerdim(self, tup)
    988         for i, key in enumerate(tup):
    989             if is_label_like(key) or isinstance(key, tuple):
--> 990                 section = self._getitem_axis(key, axis=i)
    991 
    992                 # we have yielded a scalar ?
~/build/pandas-dev/pandas/pandas/core/indexing.py in _getitem_axis(self, key, axis)
   1914         # fall thru to straight lookup
   1915         self._validate_key(key, axis)
-> 1916         return self._get_label(key, axis=axis)
   1917 
   1918 
~/build/pandas-dev/pandas/pandas/core/indexing.py in _get_label(self, label, axis)
    146             raise IndexingError('no slices here, handle elsewhere')
    147 
--> 148         return self.obj._xs(label, axis=axis)
    149 
    150     def _get_loc(self, key, axis=None):
~/build/pandas-dev/pandas/pandas/core/panel.py in xs(self, key, axis)
    864         self._consolidate_inplace()
    865         axis_number = self._get_axis_number(axis)
--> 866         new_data = self._data.xs(key, axis=axis_number, copy=False)
    867         result = self._construct_return_type(new_data)
    868         copy = new_data.is_mixed_type
~/build/pandas-dev/pandas/pandas/core/internals/managers.py in xs(self, key, axis, copy, takeable)
    835             loc = key
    836         else:
--> 837             loc = self.axes[axis].get_loc(key)
    838 
    839         slicer = [slice(None, None) for _ in range(self.ndim)]
~/build/pandas-dev/pandas/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   3147                 return self._engine.get_loc(key)
   3148             except KeyError:
-> 3149                 return self._engine.get_loc(self._maybe_cast_indexer(key))
   3150         indexer = self.get_indexer([key], method=method, tolerance=tolerance)
   3151         if indexer.ndim > 1 or indexer.size > 1:
~/build/pandas-dev/pandas/pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()
~/build/pandas-dev/pandas/pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()
~/build/pandas-dev/pandas/pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()
~/build/pandas-dev/pandas/pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()
KeyError: 'ItemA'
```

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

**Comment by jorisvandenbossche** [comment]
> If you do not mind, can I try to re-open this Pull Request with the same changes? I touched something wrong in the local git and now it shows bunch of weird commits showing on this Pull Request.

@ywpark1 The diff looks fine now? Or is there still something wrong with it? (and don't worry if there are more commits, they will get squashed when merging, so it is the diff that is important)

**Comment by datapythonista** [comment]
@ywpark1 not sure what happened exactly, but in general you don't want to modify anything in a PR once it's been approved (unless you get further feedback).

There is something that will cause some trouble in this case. When working in an issue, you should create a new local git branch. Because you worked directly in master, now you can't work in any other issue until this PR is merged. If you work in something else, you'll have to do it in master and you'll mix things, or you'll have to create a new branch from a modified master, which will contain changes unrelated to the new feature.

We could fix that with a reset, and opening a new PR, but I think it's better to try to merge this, and then you can start again.

@jorisvandenbossche as `Panel` is deprecated and soon removed, wouldn't make sense to remove that whole block about it from the pandas basics documentation? Besides giving the error, I don't think anyone reading that page should find anything about `Panel`, they'll learn things that will be removed very soon.

**Comment by jreback** [comment]
> @jorisvandenbossche as Panel is deprecated and soon removed, wouldn't make sense to remove that whole block about it from the pandas basics documentation? Besides giving the error, I don't think anyone reading that page should find anything about Panel, they'll learn things that will be removed very soon.

yes could remove that whole block is ok.

**Comment by datapythonista** [comment]
@ywpark1 can you remove the whole `Applying with a Panel` section please.

You can check it here if that makes things easier: http://pandas.pydata.org/pandas-docs/stable/basics.html#applying-with-a-panel

**Comment by ywpark1** [comment]
Thanks for your advice @datapythonista . I was working with local master branch by mistake.
I will try to work on other issues once this is merged.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Yes, good idea to just remove that section.

There is still another problem caused by removing this section (an import is missing in another file), but I will merge this and fix that in a separate PR.

**Comment by jorisvandenbossche** [comment]
@ywpark1 Thanks a lot!

**Comment by jorisvandenbossche** [comment]
> There is still another problem caused by removing this section (an import is missing in another file), but I will merge this and fix that in a separate PR.

https://github.com/pandas-dev/pandas/pull/23841

**Pull request merged**
